src/main/java/org/isf/lab/service/LabIoOperations.java

```diff
@@ -89,16 +91,26 @@ public List<LaboratoryRow> getLabRow(Integer code) throws OHServiceException {
 	 * @throws OHServiceException
 	 */
 	public List<Laboratory> getLaboratory(boolean onWeek, int pageNo, int pageSize) throws OHServiceException {
-		
 		Pageable pageable = PageRequest.of(pageNo, pageSize);
 		if (onWeek) {
 			LocalDateTime time2 = TimeTools.getDateToday24();
 			LocalDateTime time1 = time2.minusWeeks(1);
-			List<Laboratory> pagedResult = repository.findByLabDateBetweenOrderByLabDateDesc(time1, time2, pageable);
-			return pagedResult;
+			return repository.findByLabDateBetweenOrderByLabDateDesc(time1, time2, pageable).getContent();
+
+		}
+		return repository.findAll(pageable).getContent();
+	}
+	
+	public PagedResponse<Laboratory> getLaboratoryPageable(boolean onWeek, int pageNo, int pageSize) throws OHServiceException {
+		Pageable pageable = PageRequest.of(pageNo, pageSize);
+		if (onWeek) {
```

Comment:
shouldn't all the occurrences of `onWeek` really be `oneWeek`?

Reply:
I suppose yes, but some days ago in another PR you and @ArnaudFonzam discussed to "decide for one" and that one has been chosen...

Reply:
I have used onWeek on the api that is why I also use it on the core.

Reply:
yes is it you right okay let me update it on core and on the api